src/query/union/simple_union.rs

```diff
@@ -0,0 +1,113 @@
+use crate::docset::{DocSet, TERMINATED};
+use crate::postings::Postings;
+use crate::DocId;
+
+/// A `SimpleUnion` is a `DocSet` that is the union of multiple `DocSet`.
+/// Unlike `BufferedUnion`, it doesn't do any horizon precomputation.
+///
+/// For that reason SimpleUnion is a good choice for queries that skip a lot.
+pub struct SimpleUnion<TDocSet> {
+    docsets: Vec<TDocSet>,
+    doc: DocId,
+}
+
+impl<TDocSet: DocSet> SimpleUnion<TDocSet> {
+    pub(crate) fn build(mut docsets: Vec<TDocSet>) -> SimpleUnion<TDocSet> {
+        docsets.retain(|docset| docset.doc() != TERMINATED);
+        let mut docset = SimpleUnion { docsets, doc: 0 };
+
+        docset.update_current();
+
+        docset
+    }
+
+    fn update_current(&mut self) {
+        let mut next_doc = TERMINATED;
+
+        for docset in &mut self.docsets {
+            next_doc = next_doc.min(docset.doc());
+        }
+        self.doc = next_doc;
+    }
+
+    fn advance_to_next(&mut self) -> DocId {
+        let mut next_doc = TERMINATED;
+
+        for docset in &mut self.docsets {
+            if docset.doc() <= self.doc {
+                docset.advance();
+            }
+            next_doc = next_doc.min(docset.doc());
+        }
+        self.doc = next_doc;
```

Comment:
If you have a lot of docsets, I think a BinaryHeap is supposed to be faster.

Reply:
I did some tests initially with `BinaryHeap`, but it was quite slow when there were a lot of docsets. It might have been related to the size of `SegmentPostings` which I wasn't aware of yet.